File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        def binarySearchRow(curArray : List[int]) -> bool:
            left = 0
            right = len(curArray) - 1
            while left <= right:
                curMiddle = (left + right) // 2
                if curArray[curMiddle] == target:
                    return True
                elif curArray[curMiddle] < target:
                    left = curMiddle + 1
                elif curArray[curMiddle] > target:
                    right = curMiddle - 1
            return False
        
        for i in range(0, len(matrix)):
            if binarySearchRow(matrix[i]) == True:
                return True
        return False

File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int left = 0;
        int right = nums.size() - 1;
        int middle = left + (right - left) / 2;
        while (left <= right) {
            printf("Value of middle: %d\n", nums[middle]);
            if (nums[middle] == target) {
                return middle;
            }
            else if (nums[middle] < target) {
                left = middle + 1;
            }
            else {
                right = middle - 1;
            }
            middle = left + (right - left) / 2;
        }
        return -1;
    }
};
